tensorboard/plugins/profile/tf_profile_dashboard/tf-profile-dashboard.html

```diff
@@ -399,6 +439,30 @@ <h3>No profile data was found.</h3>
     detached: function() {
       this.set('_isAttached', false);
     },
+    _openCaptureProfileDialog: function() {
+      this.$.captureProfileDialog.open();
+    },
+    _shouldDisableCaptureProfileButton: function(profilerServiceAddress, profileDuration) {
+      return !profilerServiceAddress || profileDuration <= 0;
+    },
+    _captureProfile: function() {
+      const captureProfileUrl = tf_backend.addParams(
+          tf_backend.getRouter().pluginRoute('profile', '/capture_profile'),
+          {
+            service_addr: this._profilerServiceAddress,
+            is_tpu_name: this._profilerServiceAddressType == 'tpu-name',
+            duration: this._profileDuration,
+          });
+      this._requestManager.request(captureProfileUrl).then((json) => {
```

Comment:
Hmm I did not realize this soon enough. I generally am scared of holding request open for at least a duration of the profile. Any thoughts on changing this contract so FE starts, polls (presumably with some status about the profile like memory usage), and stop?

Reply:
I think it could be a future improvement. Created an issue #2013 to track it.